test/functional/api/v2/media_controller_test.rb

```diff
@@ -26,7 +26,7 @@ class Api::V2::MediaControllerTest < ActionController::TestCase
     assert_difference('Medium.count', +1) do
       post :create, { :medium => new_medium }
     end
-    assert_response :created
+    assert_response :success
```

Comment:
Ah yes, I remember now, that's http://projects.theforeman.org/issues/6882.  Up to you if you want to address this here or separately, I realise now and accept that it's mostly broken already!

Perhaps avoid checking whether the request is a POST (`request.post?`) since we have some other POST routes that aren't creates.  You might be able to do `:status => (response.new_record? ? 201 : 200)` to see if the resource was created?